src/experience/service.py

```
@@ -129,7 +165,187 @@ def _generate_tags(experience: Experience) -> str:
+        valid_tags = [tag for tag in selected_tags if tag in AVAILABLE_TAGS]
+
+        # Ensure 1-5 tags
+        if not valid_tags:
+            # Fallback: return empty list
+            return []
+        elif len(valid_tags) > 5:
+            valid_tags = valid_tags[:5]
+
+        return valid_tags
+
+    except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)